test/unit/host_test.rb

```diff
@@ -2719,6 +2719,23 @@ def to_managed!
     end
   end
 
+  context 'hostgroup defaults' do
+    setup do
+      @group1 = FactoryGirl.create(:hostgroup, :compute_profile => compute_profiles(:one))
+      @group2 = FactoryGirl.create(:hostgroup, :compute_profile => compute_profiles(:two))
+    end
+
+    test 'sets proper compute attributes on hostgroup change' do
+      host = FactoryGirl.create(:host, :managed, :compute_resource => compute_resources(:one), :hostgroup => @group1)
+      host.set_hostgroup_defaults
```

Comment:
If you want to test the difference, I think this should suffice, no need to call the first `.set_hostgroup_defaults` to get `host.compute_attributes['cpus'] == 2` even. It gets called on create since you associate hostgroup with host directly.

```ruby
host = FactoryGirl.create(:host, :managed,
                          :compute_resource => compute_resources(:one), 
                          :hostgroup => @group1)

assert_difference('host.compute_attributes["cpus"]', 2) do
  host.attributes = host.apply_inherited_attributes('hostgroup_id' => @group2.id)
  host.set_hostgroup_defaults
end
````